numpy/core/records.py

```diff
@@ -238,17 +237,14 @@ def __getattribute__(self, attr):
         res = fielddict.get(attr, None)
         if res:
             obj = self.getfield(*res[:2])
-            # if it has fields return a recarray,
-            # if it's a string ('SU') return a chararray
+            # if it has fields return a record,
             # otherwise return the object
             try:
                 dt = obj.dtype
             except AttributeError:
-                return obj
+                return obj #happens if field is Object type
             if dt.fields:
-                return obj.view(obj.__class__)
-            if dt.char in 'SU':
-                return obj.view(chararray)
+                return obj.view((record, obj.dtype.descr))
```

Comment:
No, it's right. This is actually a *type* view (`np.void.view`) , not an *ndarray* view (`np.ndarray.view`). The argument is a dtype format of the form `(base_type, descr)`  (ie, it would go in the `dtype=` argument of `np.ndarray.view`).

As an aside, `np.void.view` only accepts a single unnamed argument, while `np.ndarray.view` has named `dtype` and `type` arguments. Yet the dosctring for `np.void.view` says

> Class generic exists solely to derive numpy scalars from, and possesses,
albeit unimplemented, all the attributes of the ndarray class
so as to provide a uniform API.

The "uniform API" does not seem that consistent since the signatures of the `view` methods are different, for one. But I will admit I do not entirely understand why scalars need to duplicate the ndarray API.